packages/nx/src/command-line/graph/graph.ts

```diff
@@ -351,19 +352,29 @@ export async function generateGraph(
     affectedProjects = [];
   }
 
-  if (args.exclude) {
-    const invalidExcludes: string[] = [];
-
-    args.exclude.forEach((project) => {
-      if (!projectExists(projects, project)) {
-        invalidExcludes.push(project);
+  let excludePatterns: string[] = [];
+  if (args.exclude && args.exclude.length > 0) {
+    try {
+      // Convert projects array to the format expected by findMatchingProjects
+      const projectsMap = projects.reduce((acc, project) => {
```

Comment:
pretty sure prunedGraph.nodes is technically a projectsMap so I don't think this is needed